Caffeine `@CacheResult`: after failed `Uni`, several concurrent invocations trying to retrieve a cache value from the same missing key may result in multiple method invocations

### Describe the bug

We (Quarkus QE) use Caffeine cache and the `@CacheResult` annotation. The annotation javadoc says following:

```
If several concurrent invocations try to retrieve a cache value from the same missing key, the method will only be invoked once. The first concurrent invocation will trigger the method invocation while the subsequent concurrent invocations will wait for the end of the method invocation to get the cached result.
```

We have a test https://github.com/quarkus-qe/quarkus-test-suite/blob/c44475410b4baecd12f69e3a7f82dcf08d9666a9/cache/caffeine/src/test/java/io/quarkus/ts/cache/caffeine/cache/caffeine/UniFailureCacheIT.java#L247 where the first method invocation returns failed `Uni` and second invocation should be successful. That is, we expect exactly 2 invocations for this method https://github.com/quarkus-qe/quarkus-test-suite/blob/c44475410b4baecd12f69e3a7f82dcf08d9666a9/cache/caffeine/src/main/java/io/quarkus/ts/cache/caffeine/ReactiveFailureCacheService.java#L61. However, _sometimes_, there are more than 2 invocations recorder.

**I realize that considering I have only seen one failure, it may not be worth for you to pay attention to this. The point of this issue is to consider if you think this situation has other explanation than that there is a bug and if so, let's check code if there isn't something obvious.** I am fine with closing this if you don't see anything.

### Expected behavior

Do what javadoc says - synchronize this.

### Actual behavior

I cannot reproduce it locally, it only failed once:

- https://github.com/quarkus-qe/quarkus-test-suite/pull/2766#issuecomment-3486479039

Full run log is here https://github.com/quarkus-qe/quarkus-test-suite/actions/runs/19066576620/job/54459758072?pr=2766. And here is extract:

```
2025-11-04T11:34:09.0523096Z 11:34:09,011 INFO  [app] 11:34:07,947 Thread 'vert.x-eventloop-thread-2' ENTERING getSingleArgData for key 'shared-resource-1762256047930'
2025-11-04T11:34:09.0524405Z 11:34:09,011 INFO  [app] 11:34:07,947 Thread 'vert.x-eventloop-thread-2' -> Counter for key 'shared-resource-1762256047930' is now 1
2025-11-04T11:34:09.0525712Z 11:34:09,011 INFO  [app] 11:34:07,950 Thread 'vert.x-eventloop-thread-3' ENTERING getSingleArgData for key 'shared-resource-1762256047930'
2025-11-04T11:34:09.0527005Z 11:34:09,011 INFO  [app] 11:34:07,950 Thread 'vert.x-eventloop-thread-3' -> Counter for key 'shared-resource-1762256047930' is now 2
2025-11-04T11:34:09.0528187Z 11:34:09,011 INFO  [app] 11:34:07,951 Thread 'vert.x-eventloop-thread-3' -> SUCCESS for key 'shared-resource-1762256047930'
2025-11-04T11:34:09.0530283Z 11:34:09,011 INFO  [app] 11:34:07,947 Exception thrown during asynchronous load: java.util.concurrent.CompletionException: io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService$CacheTestException: Service temporarily unavailable
2025-11-04T11:34:09.0532290Z 11:34:09,011 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
2025-11-04T11:34:09.0533648Z 11:34:09,011 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
2025-11-04T11:34:09.0534986Z 11:34:09,011 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
2025-11-04T11:34:09.0536371Z 11:34:09,011 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
2025-11-04T11:34:09.0537767Z 11:34:09,011 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
2025-11-04T11:34:09.0539034Z 11:34:09,011 INFO  [app] 	at io.smallrye.context.CompletableFutureWrapper.<init>(CompletableFutureWrapper.java:32)
2025-11-04T11:34:09.0540553Z 11:34:09,011 INFO  [app] 	at io.smallrye.context.impl.JdkSpecificImpl.newCompletableFutureWrapper(JdkSpecificImpl.java:23)
2025-11-04T11:34:09.0541844Z 11:34:09,011 INFO  [app] 	at io.smallrye.context.JdkSpecific.newCompletableFutureWrapper(JdkSpecific.java:28)
2025-11-04T11:34:09.0543151Z 11:34:09,011 INFO  [app] 	at io.smallrye.context.SmallRyeThreadContext.withContextCapture(SmallRyeThreadContext.java:306)
2025-11-04T11:34:09.0544510Z 11:34:09,011 INFO  [app] 	at io.smallrye.context.SmallRyeThreadContext.withContextCapture(SmallRyeThreadContext.java:302)
2025-11-04T11:34:09.0545975Z 11:34:09,011 INFO  [app] 	at io.smallrye.mutiny.context.MutinyContextManagerExtension$1.apply(MutinyContextManagerExtension.java:21)
2025-11-04T11:34:09.0547514Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.context.MutinyContextManagerExtension$1.apply(MutinyContextManagerExtension.java:17)
2025-11-04T11:34:09.0548913Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.infrastructure.Infrastructure.wrapCompletableFuture(Infrastructure.java:296)
2025-11-04T11:34:09.0550614Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniSubscribeToCompletionStage.subscribe(UniSubscribeToCompletionStage.java:31)
2025-11-04T11:34:09.0552209Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.asCompletionStage(UniSubscribe.java:174)
2025-11-04T11:34:09.0553695Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.asCompletionStage(UniSubscribe.java:162)
2025-11-04T11:34:09.0554761Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.Uni.subscribeAsCompletionStage(Uni.java:142)
2025-11-04T11:34:09.0555977Z 11:34:09,012 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:120)
2025-11-04T11:34:09.0557164Z 11:34:09,012 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:114)
2025-11-04T11:34:09.0558523Z 11:34:09,012 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.lambda$computeIfAbsent$0(LocalAsyncCache.java:425)
2025-11-04T11:34:09.0560045Z 11:34:09,012 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.lambda$computeIfAbsent$0(UnboundedLocalCache.java:290)
2025-11-04T11:34:09.0561557Z 11:34:09,012 INFO  [app] 	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
2025-11-04T11:34:09.0562866Z 11:34:09,012 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.computeIfAbsent(UnboundedLocalCache.java:285)
2025-11-04T11:34:09.0564289Z 11:34:09,012 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:424)
2025-11-04T11:34:09.0565747Z 11:34:09,012 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:351)
2025-11-04T11:34:09.0567075Z 11:34:09,012 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:113)
2025-11-04T11:34:09.0568441Z 11:34:09,012 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:107)
2025-11-04T11:34:09.0569835Z 11:34:09,012 INFO  [app] 	at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
2025-11-04T11:34:09.0571553Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage.subscribe(UniCreateFromCompletionStage.java:24)
2025-11-04T11:34:09.0572936Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0574067Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.0575165Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0576121Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniEmitOn.subscribe(UniEmitOn.java:22)
2025-11-04T11:34:09.0577127Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0578264Z 11:34:09,012 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:34)
2025-11-04T11:34:09.0579437Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0580737Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.0581776Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0582803Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.withSubscriber(UniSubscribe.java:51)
2025-11-04T11:34:09.0583550Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:110)
2025-11-04T11:34:09.0584057Z 11:34:09,013 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:88)
2025-11-04T11:34:09.0584805Z 11:34:09,013 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.UniResponseHandler.handle(UniResponseHandler.java:21)
2025-11-04T11:34:09.0586108Z 11:34:09,013 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:192)
2025-11-04T11:34:09.0588038Z 11:34:09,013 INFO  [app] 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2025-11-04T11:34:09.0589395Z 11:34:09,013 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RestInitialHandler.beginProcessing(RestInitialHandler.java:48)
2025-11-04T11:34:09.0591387Z 11:34:09,013 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:23)
2025-11-04T11:34:09.0593091Z 11:34:09,013 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:10)
2025-11-04T11:34:09.0594382Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0595438Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0596545Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0597755Z 11:34:09,013 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:62)
2025-11-04T11:34:09.0599183Z 11:34:09,013 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:40)
2025-11-04T11:34:09.0600511Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0601555Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0602617Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0603873Z 11:34:09,013 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:381)
2025-11-04T11:34:09.0605432Z 11:34:09,013 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:370)
2025-11-04T11:34:09.0606737Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0607843Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0608989Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0610005Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.route(RoutingContextImpl.java:100)
2025-11-04T11:34:09.0611204Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:69)
2025-11-04T11:34:09.0612096Z 11:34:09,013 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
2025-11-04T11:34:09.0613314Z 11:34:09,013 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:87)
2025-11-04T11:34:09.0614836Z 11:34:09,014 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:70)
2025-11-04T11:34:09.0616207Z 11:34:09,014 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:187)
2025-11-04T11:34:09.0617412Z 11:34:09,014 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:161)
2025-11-04T11:34:09.0618425Z 11:34:09,014 INFO  [app] 	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
2025-11-04T11:34:09.0619309Z 11:34:09,014 INFO  [app] 	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:163)
2025-11-04T11:34:09.0620720Z 11:34:09,014 INFO  [app] 	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:174)
2025-11-04T11:34:09.0622117Z 11:34:09,014 INFO  [app] 	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
2025-11-04T11:34:09.0623257Z 11:34:09,014 INFO  [app] 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
2025-11-04T11:34:09.0623968Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.0624968Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0625794Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0626577Z 11:34:09,014 INFO  [app] 	at io.vertx.core.http.impl.Http1xUpgradeToH2CHandler.channelRead(Http1xUpgradeToH2CHandler.java:124)
2025-11-04T11:34:09.0627358Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.0628186Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0628995Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0629713Z 11:34:09,014 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.0630995Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.0632566Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0634131Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0635550Z 11:34:09,014 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
2025-11-04T11:34:09.0636845Z 11:34:09,014 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
2025-11-04T11:34:09.0638235Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.0639859Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0641688Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0643030Z 11:34:09,014 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
2025-11-04T11:34:09.0643780Z 11:34:09,014 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
2025-11-04T11:34:09.0644559Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.0645391Z 11:34:09,014 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0646208Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0646929Z 11:34:09,015 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.0647646Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.0648465Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0649282Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0650446Z 11:34:09,015 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
2025-11-04T11:34:09.0651845Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2025-11-04T11:34:09.0653699Z 11:34:09,015 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0655234Z 11:34:09,015 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
2025-11-04T11:34:09.0656533Z 11:34:09,015 INFO  [app] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
2025-11-04T11:34:09.0657766Z 11:34:09,015 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
2025-11-04T11:34:09.0658998Z 11:34:09,015 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
2025-11-04T11:34:09.0660469Z 11:34:09,015 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
2025-11-04T11:34:09.0661475Z 11:34:09,015 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2025-11-04T11:34:09.0662487Z 11:34:09,015 INFO  [app] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
2025-11-04T11:34:09.0663538Z 11:34:09,015 INFO  [app] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2025-11-04T11:34:09.0664521Z 11:34:09,015 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2025-11-04T11:34:09.0665507Z 11:34:09,015 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:840)
2025-11-04T11:34:09.0666659Z 11:34:09,015 INFO  [app] Caused by: io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService$CacheTestException: Service temporarily unavailable
2025-11-04T11:34:09.0668792Z 11:34:09,015 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService.getSingleArgData(ReactiveFailureCacheService.java:71)
2025-11-04T11:34:09.0671031Z 11:34:09,015 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass.getSingleArgData$$superforward(Unknown Source)
2025-11-04T11:34:09.0672900Z 11:34:09,015 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass$5.apply(Unknown Source)
2025-11-04T11:34:09.0674469Z 11:34:09,015 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
2025-11-04T11:34:09.0676162Z 11:34:09,015 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
2025-11-04T11:34:09.0677532Z 11:34:09,015 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:61)
2025-11-04T11:34:09.0678800Z 11:34:09,015 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:56)
2025-11-04T11:34:09.0680051Z 11:34:09,015 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:118)
2025-11-04T11:34:09.0681091Z 11:34:09,015 INFO  [app] 	... 92 more
2025-11-04T11:34:09.0682854Z 11:34:09,016 INFO  [app] 11:34:07,954 HTTP Request to /api/reactive-failure/single-arg/shared-resource-1762256047930 failed, error id: 3b79d9b2-a29f-4432-89b6-6789d82c5c19-8: io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService$CacheTestException: Service temporarily unavailable
2025-11-04T11:34:09.0685181Z 11:34:09,016 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService.getSingleArgData(ReactiveFailureCacheService.java:71)
2025-11-04T11:34:09.0686803Z 11:34:09,016 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass.getSingleArgData$$superforward(Unknown Source)
2025-11-04T11:34:09.0688255Z 11:34:09,016 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass$5.apply(Unknown Source)
2025-11-04T11:34:09.0689963Z 11:34:09,016 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
2025-11-04T11:34:09.0691635Z 11:34:09,016 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
2025-11-04T11:34:09.0693319Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:61)
2025-11-04T11:34:09.0694664Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:56)
2025-11-04T11:34:09.0695805Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:118)
2025-11-04T11:34:09.0697024Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:114)
2025-11-04T11:34:09.0698397Z 11:34:09,016 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.lambda$computeIfAbsent$0(LocalAsyncCache.java:425)
2025-11-04T11:34:09.0699917Z 11:34:09,016 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.lambda$computeIfAbsent$0(UnboundedLocalCache.java:290)
2025-11-04T11:34:09.0701531Z 11:34:09,016 INFO  [app] 	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
2025-11-04T11:34:09.0702913Z 11:34:09,016 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.computeIfAbsent(UnboundedLocalCache.java:285)
2025-11-04T11:34:09.0704404Z 11:34:09,016 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:424)
2025-11-04T11:34:09.0705806Z 11:34:09,016 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:351)
2025-11-04T11:34:09.0707057Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:113)
2025-11-04T11:34:09.0708179Z 11:34:09,016 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:107)
2025-11-04T11:34:09.0709472Z 11:34:09,016 INFO  [app] 	at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
2025-11-04T11:34:09.0711243Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage.subscribe(UniCreateFromCompletionStage.java:24)
2025-11-04T11:34:09.0712656Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0713808Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.0714491Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0715070Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniEmitOn.subscribe(UniEmitOn.java:22)
2025-11-04T11:34:09.0715626Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0716270Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:34)
2025-11-04T11:34:09.0716898Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0717518Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.0718135Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.0718696Z 11:34:09,016 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.withSubscriber(UniSubscribe.java:51)
2025-11-04T11:34:09.0719259Z 11:34:09,017 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:110)
2025-11-04T11:34:09.0719942Z 11:34:09,017 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:88)
2025-11-04T11:34:09.0721357Z 11:34:09,017 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.UniResponseHandler.handle(UniResponseHandler.java:21)
2025-11-04T11:34:09.0723138Z 11:34:09,017 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:192)
2025-11-04T11:34:09.0725367Z 11:34:09,017 INFO  [app] 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2025-11-04T11:34:09.0727090Z 11:34:09,017 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RestInitialHandler.beginProcessing(RestInitialHandler.java:48)
2025-11-04T11:34:09.0728756Z 11:34:09,017 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:23)
2025-11-04T11:34:09.0730751Z 11:34:09,017 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:10)
2025-11-04T11:34:09.0731906Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0732526Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0733436Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0734685Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:62)
2025-11-04T11:34:09.0736116Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:40)
2025-11-04T11:34:09.0737297Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0738405Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0739538Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0741146Z 11:34:09,017 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:381)
2025-11-04T11:34:09.0742768Z 11:34:09,017 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:370)
2025-11-04T11:34:09.0744146Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.0745101Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.0746232Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.0747276Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.route(RoutingContextImpl.java:100)
2025-11-04T11:34:09.0748240Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:69)
2025-11-04T11:34:09.0749075Z 11:34:09,017 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
2025-11-04T11:34:09.0750470Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:87)
2025-11-04T11:34:09.0751970Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:70)
2025-11-04T11:34:09.0753312Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:187)
2025-11-04T11:34:09.0754514Z 11:34:09,017 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:161)
2025-11-04T11:34:09.0755530Z 11:34:09,017 INFO  [app] 	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
2025-11-04T11:34:09.0756615Z 11:34:09,017 INFO  [app] 	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:163)
2025-11-04T11:34:09.0757363Z 11:34:09,017 INFO  [app] 	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:174)
2025-11-04T11:34:09.0757987Z 11:34:09,018 INFO  [app] 	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
2025-11-04T11:34:09.0758634Z 11:34:09,018 INFO  [app] 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
2025-11-04T11:34:09.0759322Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.0760393Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0761904Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.0763395Z 11:34:09,018 INFO  [app] 	at io.vertx.core.http.impl.Http1xUpgradeToH2CHandler.channelRead(Http1xUpgradeToH2CHandler.java:124)
2025-11-04T11:34:09.0764836Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.0766380Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.0768225Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1397501Z 11:34:09,018 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.1399520Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.1403055Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1406063Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1407634Z 11:34:09,018 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
2025-11-04T11:34:09.1409091Z 11:34:09,018 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
2025-11-04T11:34:09.1410770Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.1412170Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1413053Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1413752Z 11:34:09,018 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
2025-11-04T11:34:09.1414336Z 11:34:09,018 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
2025-11-04T11:34:09.1415087Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.1415917Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1416723Z 11:34:09,018 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1417433Z 11:34:09,018 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.1418140Z 11:34:09,019 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.1419261Z 11:34:09,019 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1420067Z 11:34:09,019 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1421145Z 11:34:09,019 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
2025-11-04T11:34:09.1421926Z 11:34:09,019 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2025-11-04T11:34:09.1422739Z 11:34:09,019 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1423489Z 11:34:09,019 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
2025-11-04T11:34:09.1424193Z 11:34:09,019 INFO  [app] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
2025-11-04T11:34:09.1424857Z 11:34:09,019 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
2025-11-04T11:34:09.1425495Z 11:34:09,019 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
2025-11-04T11:34:09.1426128Z 11:34:09,019 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
2025-11-04T11:34:09.1426653Z 11:34:09,019 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2025-11-04T11:34:09.1427234Z 11:34:09,019 INFO  [app] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
2025-11-04T11:34:09.1427855Z 11:34:09,019 INFO  [app] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2025-11-04T11:34:09.1428462Z 11:34:09,019 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2025-11-04T11:34:09.1428988Z 11:34:09,019 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:840)
2025-11-04T11:34:09.1430053Z 11:34:09,019 INFO  [app] 11:34:07,955 HTTP Request to /api/reactive-failure/single-arg/shared-resource-1762256047930 failed, error id: 3b79d9b2-a29f-4432-89b6-6789d82c5c19-9: io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService$CacheTestException: Service temporarily unavailable
2025-11-04T11:34:09.1431510Z 11:34:09,019 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService.getSingleArgData(ReactiveFailureCacheService.java:71)
2025-11-04T11:34:09.1432370Z 11:34:09,019 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass.getSingleArgData$$superforward(Unknown Source)
2025-11-04T11:34:09.1433120Z 11:34:09,019 INFO  [app] 	at io.quarkus.ts.cache.caffeine.ReactiveFailureCacheService_Subclass$5.apply(Unknown Source)
2025-11-04T11:34:09.1433841Z 11:34:09,019 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
2025-11-04T11:34:09.1434590Z 11:34:09,019 INFO  [app] 	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
2025-11-04T11:34:09.1435296Z 11:34:09,019 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:61)
2025-11-04T11:34:09.1435956Z 11:34:09,019 INFO  [app] 	at io.quarkus.cache.runtime.CacheResultInterceptor$1.apply(CacheResultInterceptor.java:56)
2025-11-04T11:34:09.1436613Z 11:34:09,019 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:118)
2025-11-04T11:34:09.1437268Z 11:34:09,019 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4$1.apply(CaffeineCacheImpl.java:114)
2025-11-04T11:34:09.1438017Z 11:34:09,019 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.lambda$computeIfAbsent$0(LocalAsyncCache.java:425)
2025-11-04T11:34:09.1438949Z 11:34:09,019 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.lambda$computeIfAbsent$0(UnboundedLocalCache.java:290)
2025-11-04T11:34:09.1439815Z 11:34:09,019 INFO  [app] 	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
2025-11-04T11:34:09.1440777Z 11:34:09,019 INFO  [app] 	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.computeIfAbsent(UnboundedLocalCache.java:285)
2025-11-04T11:34:09.1441571Z 11:34:09,020 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:424)
2025-11-04T11:34:09.1442368Z 11:34:09,020 INFO  [app] 	at com.github.benmanes.caffeine.cache.LocalAsyncCache$AsyncAsMapView.computeIfAbsent(LocalAsyncCache.java:351)
2025-11-04T11:34:09.1443082Z 11:34:09,020 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:113)
2025-11-04T11:34:09.1443716Z 11:34:09,020 INFO  [app] 	at io.quarkus.cache.runtime.caffeine.CaffeineCacheImpl$4.get(CaffeineCacheImpl.java:107)
2025-11-04T11:34:09.1444404Z 11:34:09,020 INFO  [app] 	at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
2025-11-04T11:34:09.1445232Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage.subscribe(UniCreateFromCompletionStage.java:24)
2025-11-04T11:34:09.1445995Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.1446628Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.1447241Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.1447784Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniEmitOn.subscribe(UniEmitOn.java:22)
2025-11-04T11:34:09.1448334Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.1448977Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:34)
2025-11-04T11:34:09.1449603Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.1450343Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2025-11-04T11:34:09.1450968Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
2025-11-04T11:34:09.1451528Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.withSubscriber(UniSubscribe.java:51)
2025-11-04T11:34:09.1452076Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:110)
2025-11-04T11:34:09.1452585Z 11:34:09,020 INFO  [app] 	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:88)
2025-11-04T11:34:09.1453215Z 11:34:09,020 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.UniResponseHandler.handle(UniResponseHandler.java:21)
2025-11-04T11:34:09.1454202Z 11:34:09,020 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:192)
2025-11-04T11:34:09.1455271Z 11:34:09,020 INFO  [app] 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2025-11-04T11:34:09.1456155Z 11:34:09,020 INFO  [app] 	at org.jboss.resteasy.reactive.server.handlers.RestInitialHandler.beginProcessing(RestInitialHandler.java:48)
2025-11-04T11:34:09.1457031Z 11:34:09,020 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:23)
2025-11-04T11:34:09.1457932Z 11:34:09,020 INFO  [app] 	at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:10)
2025-11-04T11:34:09.1458708Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.1459361Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.1459973Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.1460860Z 11:34:09,020 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:62)
2025-11-04T11:34:09.1461657Z 11:34:09,020 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:40)
2025-11-04T11:34:09.1462309Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.1462895Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.1463500Z 11:34:09,020 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.1464214Z 11:34:09,020 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:381)
2025-11-04T11:34:09.1465092Z 11:34:09,021 INFO  [app] 	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$13.handle(ResteasyReactiveRecorder.java:370)
2025-11-04T11:34:09.1465770Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
2025-11-04T11:34:09.1466356Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2025-11-04T11:34:09.1466959Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
2025-11-04T11:34:09.1467507Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RoutingContextImpl.route(RoutingContextImpl.java:100)
2025-11-04T11:34:09.1468015Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:69)
2025-11-04T11:34:09.1468470Z 11:34:09,021 INFO  [app] 	at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
2025-11-04T11:34:09.1469096Z 11:34:09,021 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:87)
2025-11-04T11:34:09.1469893Z 11:34:09,021 INFO  [app] 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:70)
2025-11-04T11:34:09.1470781Z 11:34:09,021 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:187)
2025-11-04T11:34:09.1471405Z 11:34:09,021 INFO  [app] 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:161)
2025-11-04T11:34:09.1471936Z 11:34:09,021 INFO  [app] 	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
2025-11-04T11:34:09.1472419Z 11:34:09,021 INFO  [app] 	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:163)
2025-11-04T11:34:09.1473033Z 11:34:09,021 INFO  [app] 	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:174)
2025-11-04T11:34:09.1473643Z 11:34:09,021 INFO  [app] 	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
2025-11-04T11:34:09.1474157Z 11:34:09,021 INFO  [app] 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
2025-11-04T11:34:09.1474836Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.1475657Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1476460Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1477236Z 11:34:09,021 INFO  [app] 	at io.vertx.core.http.impl.Http1xUpgradeToH2CHandler.channelRead(Http1xUpgradeToH2CHandler.java:124)
2025-11-04T11:34:09.1478138Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.1478973Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1479869Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1480683Z 11:34:09,021 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.1481424Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.1482241Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1483048Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1483792Z 11:34:09,021 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
2025-11-04T11:34:09.1484474Z 11:34:09,021 INFO  [app] 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
2025-11-04T11:34:09.1485208Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.1486023Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1486849Z 11:34:09,021 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1487526Z 11:34:09,022 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
2025-11-04T11:34:09.1488117Z 11:34:09,022 INFO  [app] 	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
2025-11-04T11:34:09.1488838Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2025-11-04T11:34:09.1489658Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1490561Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1491273Z 11:34:09,022 INFO  [app] 	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
2025-11-04T11:34:09.1491986Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2025-11-04T11:34:09.1492804Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1493609Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2025-11-04T11:34:09.1494380Z 11:34:09,022 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
2025-11-04T11:34:09.1495160Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2025-11-04T11:34:09.1495976Z 11:34:09,022 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2025-11-04T11:34:09.1496729Z 11:34:09,022 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
2025-11-04T11:34:09.1497422Z 11:34:09,022 INFO  [app] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
2025-11-04T11:34:09.1498193Z 11:34:09,022 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
2025-11-04T11:34:09.1498818Z 11:34:09,022 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
2025-11-04T11:34:09.1499437Z 11:34:09,022 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
2025-11-04T11:34:09.1500066Z 11:34:09,022 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2025-11-04T11:34:09.1500752Z 11:34:09,022 INFO  [app] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
2025-11-04T11:34:09.1501385Z 11:34:09,022 INFO  [app] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2025-11-04T11:34:09.1501992Z 11:34:09,022 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2025-11-04T11:34:09.1502515Z 11:34:09,022 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:840)
2025-11-04T11:34:09.1503080Z 11:34:09,022 INFO  [app] 11:34:07,955 Thread 'vert.x-eventloop-thread-0' ENTERING getSingleArgData for key 'shared-resource-1762256047930'
2025-11-04T11:34:09.1503771Z 11:34:09,022 INFO  [app] 11:34:07,956 Thread 'vert.x-eventloop-thread-0' -> Counter for key 'shared-resource-1762256047930' is now 3
2025-11-04T11:34:09.1504410Z 11:34:09,022 INFO  [app] 11:34:07,956 Thread 'vert.x-eventloop-thread-0' -> SUCCESS for key 'shared-resource-1762256047930'
2025-11-04T11:34:10.1051098Z 11:34:10,100 INFO  [app] Service stopped (Quarkus JVM mode)
2025-11-04T11:34:10.1271897Z [ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.340 s <<< FAILURE! -- in io.quarkus.ts.cache.caffeine.cache.caffeine.UniFailureCacheIT
2025-11-04T11:34:10.1274117Z [ERROR] io.quarkus.ts.cache.caffeine.cache.caffeine.UniFailureCacheIT.testConcurrentAccessAndRecovery -- Time elapsed: 0.080 s <<< FAILURE!
2025-11-04T11:34:10.1276184Z org.opentest4j.AssertionFailedError: Cache lock should prevent excessive calls. Expected <= 2, but was 3 ==> expected: <true> but was: <false>
2025-11-04T11:34:10.1277730Z 	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
2025-11-04T11:34:10.1279150Z 	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
2025-11-04T11:34:10.1280661Z 	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
2025-11-04T11:34:10.1281745Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
2025-11-04T11:34:10.1282910Z 	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:214)
2025-11-04T11:34:10.1284552Z 	at io.quarkus.ts.cache.caffeine.cache.caffeine.UniFailureCacheIT.testConcurrentAccessAndRecovery(UniFailureCacheIT.java:270)
2025-11-04T11:34:10.1286115Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
2025-11-04T11:34:10.1287105Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2025-11-04T11:34:10.1288616Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```


### How to Reproduce?

You can run this test with following tests:

```
git clone git@github.com:quarkus-qe/quarkus-test-suite.git
cd quarkus-test-suite/cache/caffeine
mvn clean verify -Dtest=UniFailureCacheIT#testConcurrentAccessAndRecovery -Dreruns=0
```

But I doubt you will reproduce it.

### Output of `uname -a` or `ver`

Ubuntu latest

### Output of `java -version`

Temurin 21

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9.9

### Additional information

@jcarranzan is contact person on this issue

I wonder if it could be related to this comment https://github.com/quarkusio/quarkus/pull/39762/changes#r1562065918.

--- a/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
+++ b/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
@@ -2,6 +2,18 @@ package io.quarkus.cache.test.runtime;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
@@ -17,10 +29,14 @@ import io.vertx.core.impl.NoStackTraceException;
 public class UniReturnTypeWithFailureTest {
 
     @RegisterExtension
-    static final QuarkusUnitTest TEST = new QuarkusUnitTest().withApplicationRoot((jar) -> jar.addClass(CachedService.class));
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar.addClass(CachedService.class).addClass(ConcurrentFailureService.class));
 
     @Inject
     CachedService cachedService;
+
+    @Inject
+    ConcurrentFailureService concurrentFailureService;
 
     @Test
     void testCacheResult() {
@@ -30,6 +46,56 @@ public class UniReturnTypeWithFailureTest {
         assertEquals(2, cachedService.getCacheResultInvocations());
         assertEquals("", cachedService.cacheResult("k1").await().indefinitely());
         assertEquals(2, cachedService.getCacheResultInvocations());
+    }
+
+    /**
+     * Reproducer for #51928: After a failed Uni, several concurrent invocations
+     * may result in multiple method invocations instead of being synchronized.
+     */
+    @Test
+    void testConcurrentAccessAndRecovery() throws InterruptedException {
+        String key = "retry-resource-" + System.currentTimeMillis();
+        int numberOfConcurrentThreads = 30;
+        ExecutorService executor = Executors.newFixedThreadPool(numberOfConcurrentThreads);
+        CyclicBarrier startBarrier = new CyclicBarrier(numberOfConcurrentThreads + 1);
+
+        concurrentFailureService.resetCounter(key);
+
+        for (int i = 0; i < numberOfConcurrentThreads; i++) {
+            executor.submit(() -> {
+                try {
+                    startBarrier.await();
+                    try {
+                        concurrentFailureService.getSingleArgData(key).await().indefinitely();
+                    } catch (NoStackTraceException e) {
+                        try {
+                            concurrentFailureService.getSingleArgData(key).await().indefinitely();
+                        } catch (Exception e2) {
+                            // Ignore
+                        }
+                    }
+                } catch (Exception e) {
+                    // Ignore
+                }
+            });
+        }
+
+        try {
+            startBarrier.await(5, TimeUnit.SECONDS);
+        } catch (BrokenBarrierException | TimeoutException e) {
+            // Ignore
+        }
+        executor.shutdown();
+        executor.awaitTermination(10, TimeUnit.SECONDS);
+
+        int invocations = concurrentFailureService.getInvocations(key);
+        if (invocations > 2) {
+            assertTrue(false,
+                    String.format("Expected <= 2, but was %d for key %s. " +
+                            "This reproduces issue #51928 - multiple concurrent invocations after " +
+                            "failed Uni are not properly synchronized.",
+                            invocations, key));
+        }
     }
 
     @ApplicationScoped
@@ -48,6 +114,34 @@ public class UniReturnTypeWithFailureTest {
 
         public int getCacheResultInvocations() {
             return cacheResultInvocations;
+        }
+    }
+
+    @ApplicationScoped
+    static class ConcurrentFailureService {
+
+        private final Map<String, AtomicInteger> counters = new ConcurrentHashMap<>();
+
+        public void resetCounter(String key) {
+            counters.put(key, new AtomicInteger(0));
+        }
+
+        public int getInvocations(String key) {
+            return counters.getOrDefault(key, new AtomicInteger(0)).get();
+        }
+
+        @CacheResult(cacheName = "concurrent-test-cache")
+        public Uni<String> getSingleArgData(String key) {
+            AtomicInteger counter = counters.computeIfAbsent(key, k -> new AtomicInteger(0));
+            int invocationNumber = counter.incrementAndGet();
+
+            if (invocationNumber == 1) {
+                return Uni.createFrom().item("delay")
+                        .onItem().delayIt().by(Duration.ofMillis(200))
+                        .onItem().transformToUni(s -> Uni.createFrom().failure(new NoStackTraceException("First call fails")));
+            } else {
+                return Uni.createFrom().item("success-" + key);
+            }
         }
     }
 }

--- a/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
+++ b/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
@@ -92,6 +92,29 @@ public class UniReturnTypeWithFailureTest {
                 "Timeout failures should not be cached - method should be invoked again (issue #39677)");
     }
 
+    @Test
+    void testSuccessfulCachingAfterFailure() {
+        String key = "recovery-test-key";
+        failureCachingService.resetCounter(key);
+
+        assertThrows(NoStackTraceException.class,
+                () -> failureCachingService.getUsernameByIdWithRecovery(key).await().indefinitely());
+        assertEquals(1, failureCachingService.getInvocations(key));
+
+        assertThrows(NoStackTraceException.class,
+                () -> failureCachingService.getUsernameByIdWithRecovery(key).await().indefinitely());
+        assertEquals(2, failureCachingService.getInvocations(key));
+
+        String result = failureCachingService.getUsernameByIdWithRecovery(key).await().indefinitely();
+        assertEquals("username-" + key, result);
+        assertEquals(3, failureCachingService.getInvocations(key));
+
+        String cachedResult = failureCachingService.getUsernameByIdWithRecovery(key).await().indefinitely();
+        assertEquals("username-" + key, cachedResult);
+        assertEquals(3, failureCachingService.getInvocations(key),
+                "Successful result should be cached - method should not be invoked again");
+    }
+
     /**
      * Reproducer for #51928: After a failed Uni, several concurrent invocations
      * may result in multiple method invocations instead of being synchronized.
@@ -216,6 +239,16 @@ public class UniReturnTypeWithFailureTest {
             return Uni.createFrom().item("delayed")
                     .onItem().delayIt().by(Duration.ofMillis(500))
                     .onItem().transform(s -> "username-" + userId);
+        }
+
+        @CacheResult(cacheName = "failure-cache")
+        public Uni<String> getUsernameByIdWithRecovery(String userId) {
+            AtomicInteger counter = counters.computeIfAbsent(userId, k -> new AtomicInteger(0));
+            int invocationNumber = counter.incrementAndGet();
+            if (invocationNumber <= 2) {
+                return Uni.createFrom().failure(new NoStackTraceException("Error when getUsername"));
+            }
+            return Uni.createFrom().item("username-" + userId);
         }
     }
 }

--- a/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
+++ b/extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/UniReturnTypeWithFailureTest.java
@@ -30,13 +30,17 @@ public class UniReturnTypeWithFailureTest {
 
     @RegisterExtension
     static final QuarkusUnitTest TEST = new QuarkusUnitTest()
-            .withApplicationRoot((jar) -> jar.addClass(CachedService.class).addClass(ConcurrentFailureService.class));
+            .withApplicationRoot((jar) -> jar.addClass(CachedService.class).addClass(ConcurrentFailureService.class)
+                    .addClass(FailureCachingService.class));
 
     @Inject
     CachedService cachedService;
 
     @Inject
     ConcurrentFailureService concurrentFailureService;
+
+    @Inject
+    FailureCachingService failureCachingService;
 
     @Test
     void testCacheResult() {
@@ -46,6 +50,46 @@ public class UniReturnTypeWithFailureTest {
         assertEquals(2, cachedService.getCacheResultInvocations());
         assertEquals("", cachedService.cacheResult("k1").await().indefinitely());
         assertEquals(2, cachedService.getCacheResultInvocations());
+    }
+
+    /**
+     * Reproducer for #39677: Failures are cached instead of being retried.
+     */
+    @Test
+    void testFailureCaching() {
+        String key = "failure-test-key";
+        failureCachingService.resetCounter(key);
+
+        assertThrows(NoStackTraceException.class,
+                () -> failureCachingService.getUsernameById(key).await().indefinitely());
+        assertEquals(1, failureCachingService.getInvocations(key));
+
+        assertThrows(NoStackTraceException.class,
+                () -> failureCachingService.getUsernameById(key).await().indefinitely());
+
+        assertEquals(2, failureCachingService.getInvocations(key),
+                "Failures should not be cached - method should be invoked again (issue #39677)");
+    }
+
+    /**
+     * Reproducer for #39677: Timeout failures are cached instead of being retried.
+     */
+    @Test
+    void testTimeoutFailureCaching() {
+        String key = "timeout-test-key";
+        failureCachingService.resetCounter(key);
+
+        assertThrows(io.smallrye.mutiny.TimeoutException.class,
+                () -> failureCachingService.getUsernameByIdWithTimeout(key).await()
+                        .atMost(Duration.ofMillis(100)));
+        assertEquals(1, failureCachingService.getInvocations(key));
+
+        assertThrows(io.smallrye.mutiny.TimeoutException.class,
+                () -> failureCachingService.getUsernameByIdWithTimeout(key).await()
+                        .atMost(Duration.ofMillis(100)));
+
+        assertEquals(2, failureCachingService.getInvocations(key),
+                "Timeout failures should not be cached - method should be invoked again (issue #39677)");
     }
 
     /**
@@ -142,6 +186,36 @@ public class UniReturnTypeWithFailureTest {
             } else {
                 return Uni.createFrom().item("success-" + key);
             }
+        }
+    }
+
+    @ApplicationScoped
+    static class FailureCachingService {
+
+        private final Map<String, AtomicInteger> counters = new ConcurrentHashMap<>();
+
+        public void resetCounter(String key) {
+            counters.put(key, new AtomicInteger(0));
+        }
+
+        public int getInvocations(String key) {
+            return counters.getOrDefault(key, new AtomicInteger(0)).get();
+        }
+
+        @CacheResult(cacheName = "failure-cache")
+        public Uni<String> getUsernameById(String userId) {
+            AtomicInteger counter = counters.computeIfAbsent(userId, k -> new AtomicInteger(0));
+            counter.incrementAndGet();
+            return Uni.createFrom().failure(new NoStackTraceException("Error when getUsername"));
+        }
+
+        @CacheResult(cacheName = "timeout-failure-cache")
+        public Uni<String> getUsernameByIdWithTimeout(String userId) {
+            AtomicInteger counter = counters.computeIfAbsent(userId, k -> new AtomicInteger(0));
+            counter.incrementAndGet();
+            return Uni.createFrom().item("delayed")
+                    .onItem().delayIt().by(Duration.ofMillis(500))
+                    .onItem().transform(s -> "username-" + userId);
         }
     }
 }

--- a/extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CacheResultInterceptor.java
+++ b/extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CacheResultInterceptor.java
@@ -65,11 +65,6 @@ public class CacheResultInterceptor extends CacheInterceptor {
                             throw new CacheException(e);
                         }
                     }
-                }).onFailure().call(new Function<>() {
-                    @Override
-                    public Uni<?> apply(Throwable throwable) {
-                        return cache.invalidate(key).replaceWith(throwable);
-                    }
                 });
 
                 if (binding.lockTimeout() <= 0) {

